curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/64110998bc00321fd8052ab5.md

```diff
@@ -0,0 +1,374 @@
+---
+id: 64110998bc00321fd8052ab5
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+The last sorting algorithm you will implement is the <dfn>insertion sort</dfn>. This algorithm works by building up a sorted array at the beginning of the list. It begins the sorted array with the first element. Then it inspects the next element and swaps it backwards into the sorted array until it is in sorted position, and so on.
+
+Start by declaring an `insertionSort` variable and assigning it an arrow function which takes an `array` parameter.
+
+# --hints--
+
+You should use `const` to declare an `insertionSort` variable.
+
+```js
+assert.match(code, /const\s+insertionSort\s*=/);
+```
+
+`insertionSort` should be a function.
+
+```js
+assert.isFunction(insertionSort);
+```
+
+`insertionSort` should use arrow syntax.
```

Comment:
Campers should be able to pass with this code without parenthesis for the `array` parameter

```
const insertionSort = array => {}

```